### PR TITLE
BaSP-MR-112: Activities redux post put logic

### PR DESCRIPTION
**Pull request by lorenzomauro59**
- Replace the data from the local state
- Move the logic of the post to the thunk file in the activities folder and apply it to the entity index file.
- Move the logic of the put to the thunk file in the activities folder and apply it to the entity index file.
- To make the user experience more pleasant, for each request, a spinner (isLoading) should be implemented so that the user understands that a request is being made, for example when loading the table.

### CODE REVIEW

**Comment by lorenzomauro59** [comment]
[[Activities] - Move the post and put logic.](https://trello.com/c/toPC9ISi/112-activities-move-the-post-and-put-logic)

File: src/Components/Activities/Form/index.jsx [hunk 1]
```diff
@@ -18,90 +19,70 @@ const Form = () => {
...
+      activitiesById(id);
+    }
+  }, [id]);
+
+  const activitiesById = (id) => {
+    const activity = activities.find((activity) => activity._id === id);
+    if (activity) {
+      setActivity(activity);
+    } else {
+      console.error('Member not found');
```
Comment (EmaBaldor) [inline src/Components/Activities/Form/index.jsx]: Check this line.

File: src/Components/Activities/index.jsx [hunk 2]
```diff
@@ -48,7 +50,7 @@ const Activities = () => {
       setBodyModal(data.message);
       setTypeStyle('success');
       setIsDelete(false);
-      setReload(true);
+      refreshTable();
```
Comment (EmaBaldor) [inline src/Components/Activities/index.jsx]: If you use the filter in reducer Delete you don't need to call refreshTable again.

Comment (lorenzomauro59) [reply src/Components/Activities/index.jsx]: I already tried it, however if I take it out it does not update the list anyway

File: src/Redux/activities/actions.js [hunk 3]
```diff
@@ -7,7 +7,13 @@ import {
...
   GET_ACTIVITIES_BY_ID_SUCCESS,
-  GET_ACTIVITIES_BY_ID_ERROR
+  GET_ACTIVITIES_BY_ID_ERROR,
+  PUT_ACTIVITIES_PENDING,
+  PUT_ACTIVITIES_SUCCESS,
+  PUT_ACTIVITIES_ERROR,
+  POST_ACTIVITIES_PENDING,
+  POST_ACTIVITIES_SUCCESS,
+  POST_ACTIVITIES_ERROR
 } from './constants';
```
Comment (marcohzo) [inline src/Redux/activities/actions.js]: You can change this by:
```
import * as types from './constants';
```

File: src/Redux/activities/actions.js [hunk 4]
```diff
@@ -69,3 +75,43 @@ export const getActivitiesByIdError = (error) => {
     payload: error
   };
 };
+
+export const putActivitiesPending = () => {
+  return {
+    type: PUT_ACTIVITIES_PENDING
```
Comment (marcohzo) [inline src/Redux/activities/actions.js]: if you follow the change mentioned above you should add "types" before each constant
```
    type: types.PUT_ACTIVITIES_PENDING
```

File: src/Components/Activities/Form/index.jsx [hunk 5]
```diff
@@ -18,90 +19,70 @@ const Form = () => {
...
-      setActivity({ name: data.name, description: data.description });
-    } catch (error) {
-      console.error(error);
+  useEffect(() => {
+    if (id) {
+      activitiesById(id);
+    }
+  }, [id]);
+
+  const activitiesById = (id) => {
```
Comment (ale-schmidt) [inline src/Components/Activities/Form/index.jsx]: I think is better to call this function as activitieById in singular, what do you think?

File: src/Components/Activities/Form/index.jsx [hunk 6]
```diff
@@ -4,11 +4,12 @@ import Input from '../Input';
 import Button from '../../Shared/Button';
 import SharedModal from '../../Shared/Modal';
 import styles from './form.module.css';
+import { putActivities, postActivities } from '../../../Redux/activities/thunks';
```
Comment (ale-schmidt) [inline src/Components/Activities/Form/index.jsx:7]: also here naming the functions I think should be in singular (avtivitie), let me know what you think

**Review by EmaBaldor — APPROVED** [review]
Good Job!

File: src/Redux/activities/actions.js (hunk 4 above)
Comment (lorenzomauro59) [reply src/Redux/activities/actions.js]: thanks marco!! will try !

File: src/Components/Activities/Form/index.jsx [hunk 7]
```diff
@@ -18,90 +19,70 @@ const Form = () => {
...
-      const response = await fetch(`${process.env.REACT_APP_API_URL}/api/activities/${id}`);
-      const { data } = await response.json();
-      setActivity({ name: data.name, description: data.description });
-    } catch (error) {
-      console.error(error);
+  useEffect(() => {
+    if (id) {
+      activitiesById(id);
+    }
+  }, [id]);
```
Comment (ale-schmidt) [inline src/Components/Activities/Form/index.jsx]: you could take out the "id" param from the useEffect function

**Review by ale-schmidt — CHANGES_REQUESTED** [review]

File: src/Components/Activities/index.jsx (hunk 2 above)
Comment (lorenzomauro59) [reply src/Components/Activities/index.jsx]: resolved!

File: src/Components/Activities/Form/index.jsx (hunk 6 above)
Comment (lorenzomauro59) [reply src/Components/Activities/Form/index.jsx:7]: I think I already put in get delete put and post 'activities', If i change 1 I must change all titles, and the getActivitiesById in thunks is already used in classes for example and can create conflict right now. thanks for the advice!!

File: src/Components/Activities/Form/index.jsx (hunk 5 above)
Comment (ale-schmidt) [reply src/Components/Activities/Form/index.jsx]: no problem

**Review by ale-schmidt — APPROVED** [review]
great!

**Comment by araefedg** [comment]
The applied functionality works! nice job!

**Pull request merged**